drivers/soundwire/intel.c

```diff
@@ -883,16 +939,31 @@ static int intel_trigger(struct snd_pcm_substream *substream, int cmd,
 	}
 
 	switch (cmd) {
+	case SNDRV_PCM_TRIGGER_RESUME:
+		dev_err(dai->dev, "%s: %s: resume\n", __func__, dai->name);
+		/* fallthrough */
 	case SNDRV_PCM_TRIGGER_START:
 	case SNDRV_PCM_TRIGGER_PAUSE_RELEASE:
-	case SNDRV_PCM_TRIGGER_RESUME:
 		ret = sdw_enable_stream(dma->stream);
 		break;
 
-	case SNDRV_PCM_TRIGGER_PAUSE_PUSH:
 	case SNDRV_PCM_TRIGGER_SUSPEND:
+		dev_err(dai->dev, "%s: %s: suspend\n", __func__, dai->name);
+		dma->suspend_reinitialize = true;
```

Comment:
@plbossart SNDRV_PCM_TRIGGER_SUSPEND is not always triggered. For example. SNDRV_PCM_TRIGGER_SUSPEND will not be triggered If we do aplay -> pulse -> suspend -> resume -> release. However, we have to reinitialize in this case since the system is suspended. So my suggestion is to set it in intel_suspend().